src/pg/mod.rs

```diff
@@ -367,10 +366,35 @@ impl AsyncPgConnection {
         //
         // We apply this workaround to prevent requiring all the diesel
         // serialization code to beeing async
-        let mut metadata_lookup = PgAsyncMetadataLookup::new();
+        let mut bind_collector_0 = RawBytesBindCollector::<diesel::pg::Pg>::new();
+        let collect_bind_result_0 = query.collect_binds(
+            &mut bind_collector_0,
+            &mut SameOidEveryTime { first_byte: 0 },
+            &Pg,
+        );
+
+        let mut bind_collector_1 = RawBytesBindCollector::<diesel::pg::Pg>::new();
+        let collect_bind_result_1 = query.collect_binds(
+            &mut bind_collector_1,
+            &mut SameOidEveryTime { first_byte: 1 },
+            &Pg,
+        );
+
+        let mut metadata_lookup = PgAsyncMetadataLookup::new(&bind_collector_0);
         let collect_bind_result =
             query.collect_binds(&mut bind_collector, &mut metadata_lookup, &Pg);
 
+        let fake_oid_locations = std::iter::zip(bind_collector_0.binds, bind_collector_1.binds)
```

Comment:
I'm not sure if it might be worth to do that in the future anyway just do allow us to perform the comparison and the replacement in one pass.